crates/ra_parser/src/grammar/items.rs

```diff
@@ -107,6 +107,10 @@ pub(super) fn maybe_item(p: &mut Parser, m: Marker, flavor: ItemFlavor) -> Resul
         p.bump_remap(T![default]);
         has_mods = true;
     }
+    if p.at(IDENT) && p.at_contextual_kw("existential") && p.nth(1) == T![existential] {
```

Comment:
```suggestion
    if p.at(IDENT) && p.at_contextual_kw("existential") && p.nth(1) == T![type] {
```

I think we should look ahead for `type` here